q2_chemistree/_match.py

```diff
@@ -0,0 +1,46 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2018, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import biom
+import pandas as pd
+from skbio import TreeNode
+
+
+def match_table(tree: TreeNode,
+                feature_table: pd.DataFrame) -> biom.Table:
+    '''
+    This function filters the feature table to retain the features present in
+        the tree
+
+    Parameters
+    ----------
+    tree : TreeNode
+        skbio TreeNode object representing tree of relatedness
+        between molecules
+    feature_table : pandas dataframe
+            MS1 feature table from MZmine2 with features in rows and
+            samples in columns
+    Returns:
+    ----------
+    filtered_feature_table : pandas dataframe
+        filtered MS1 feature table that contains only the features present in
+        the tree
+    '''
+    if feature_table.shape == (0, 0):
+        raise ValueError("Cannot have empty feature table")
+    if '#row ID' not in feature_table.columns:
+        raise RuntimeError('Feature table needs to be formatted correctly')
```

Comment:
[`RuntimeError`](https://docs.python.org/3/library/exceptions.html#RuntimeError) seems surprising. Wouldn't this be better described as a `ValueError` or `KeyError` (or a custom exception)? `RuntimeErrors` aren't usually that commonly used...